app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt

```diff
@@ -284,14 +287,45 @@ class WeekFragment : Fragment(), WeeklyCalendar {
                                             ).seconds()
                                         val currentEventDuration = event.endTS - event.startTS
                                         val endTime = startTime + currentEventDuration
-                                        context?.eventsHelper?.updateEvent(
-                                            it.copy(
-                                                startTS = startTime,
-                                                endTS = endTime,
-                                                flags = it.flags.removeBit(FLAG_ALL_DAY)
-                                            ), updateAtCalDAV = true, showToasts = false
-                                        ) {
-                                            updateCalendar()
+                                        val newEvent = event.copy(
+                                            startTS = startTime,
+                                            endTS = endTime,
+                                            flags = event.flags.removeBit(FLAG_ALL_DAY)
+                                        )
+                                        if (event.repeatInterval > 0) {
+                                            val activity = this.activity as SimpleActivity
+                                            activity.runOnUiThread {
+                                                EditRepeatingEventDialog(activity) {
+                                                    activity.hideKeyboard()
+                                                    when (it) {
+                                                        null -> {
+                                                            // force update by removing hash
+                                                            lastHash = 0
```

Comment:
As we are doing this in the background, the event disappears because we already returned `true` in `setOnDragListener` which caused the view to be hidden by this code:
https://github.com/SimpleMobileTools/Simple-Calendar/blob/b2236a2a03f3772290bf26d91d3a15d25dd53c53/app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt#L897-L903

If we address this concurrency issue properly or if we choose to always force refresh when dragging events (repeating and normal) while doing drag-and-drop, #2070 will be resolved.

Reply:
I've fixed it by not nulling `currentlyDraggedView` here, but in the action itself. Can you check if it's ok?

Reply:
Nice! It's fixed now.